allennlp/predictors/predictor.py

```diff
@@ -110,13 +112,17 @@ def get_gradients(self, instances: List[Instance]) -> Tuple[Dict[str, Any], Dict
 
         dataset = Batch(instances)
         dataset.index_instances(self._model.vocab)
-        outputs = self._model.decode(
-            self._model.forward(**dataset.as_tensor_dict())  # type: ignore
-        )
-
-        loss = outputs["loss"]
-        self._model.zero_grad()
-        loss.backward()
+        # To bypass "RuntimeError: cudnn RNN backward can only be called in training mode"
+        with backends.cudnn.flags(enabled=False):
```

Comment:
This seems sketchy to me. Doesn't that disable running on the GPU altogether?

I assume you want to do this on a model that's in evaluation mode because you care about getting the gradients, but you don't care about making an update to the model. Doesn't PyTorch have a prescribed solution to this problem? It seems to me that this is just a safety check?

Reply:
Apparently this is an open issue in PyTorch: https://github.com/pytorch/pytorch/issues/10006. That thread suggests a different workaround, but yours seems safer.